wings/market/coinbase_pro_market.pyx

```diff
@@ -0,0 +1,819 @@
+        url = f"{self.COINBASE_API_ENDPOINT}{path_url}"
+        try:
+            [cancelled_id] = await self._api_request("delete", path_url=path_url)
+            if cancelled_id == exchange_order_id:
+                self.logger().info(f"Successfully cancelled order {order_id}.")
+                self.c_stop_tracking_order(order_id)
+                self.c_trigger_event(self.MARKET_ORDER_CANCELLED_EVENT_TAG,
+                                     OrderCancelledEvent(self._current_timestamp, order_id))
+        except Exception as e:
+            self.logger().error(f"Failed to cancel order {order_id}: {str(e)}")
+        return order_id
+
+    cdef c_cancel(self, str symbol, str order_id):
+        asyncio.ensure_future(self.execute_cancel(symbol, order_id))
+        return order_id
+
+    async def cancel_all(self, timeout_seconds: float) -> List[CancellationResult]:
+        incomplete_orders = [o for o in self._in_flight_orders.values() if not o.is_done]
+        successful_cancellations = []
+
+        try:
+            path_url = "/orders"
+            url = f"{self.COINBASE_API_ENDPOINT}{path_url}"
+            async with timeout(timeout_seconds):
+                results = await self._api_request("delete", path_url=path_url)
+
+            exchange_order_id_map = dict(zip([o.exchange_order_id for o in incomplete_orders], incomplete_orders))
+            for exchange_order_id in results:
+                order = exchange_order_id_map.get(exchange_order_id)
+                if order is not None:
+                    exchange_order_id_map.pop(exchange_order_id, None)
+                    successful_cancellations.append(CancellationResult(order.client_order_id, True))
+        except Exception:
+            self.logger().error(f"Unexpected error cancelling orders.", exc_info=True)
+
+        failed_cancellations = [CancellationResult(order.client_order_id, False)
+                                for order in list(exchange_order_id_map.values())]
+        return successful_cancellations + failed_cancellations
+
+    async def _status_polling_loop(self):
+        while True:
+            try:
+                self._poll_notifier = asyncio.Event()
+                await self._poll_notifier.wait()
+
+                await asyncio.gather(
+                    self._update_balances(),
+                    self._update_trading_rules(),
+                    self._update_order_status(),
+                    self._update_eth_tx_status(),
+                )
+            except asyncio.CancelledError:
+                raise
+            except Exception:
+                self.logger().error("Unexpected error while fetching account updates.", exc_info=True)
+
+    async def get_order(self, client_order_id: str) -> Dict[str, any]:
+        order = self._in_flight_orders.get(client_order_id)
+        exchange_order_id = await order.get_exchange_order_id()
+        path_url = f"/orders/{exchange_order_id}"
```

Comment:
use `https://docs.pro.coinbase.com/#list-orders` instead of checking each individual order

Reply:
@chowryan Tried the list_order approach and it is working. But not sure if it is better because the endpoint returns the status of every order ever made, which can be quite a lot of data that we have to iterate through every interval.

Reply:
Note the doc says the request is paginated. What that means is it'll return at most 100 orders in the first page.

Reply:
@martin Made the change and it is now working with list_orders. Ryan and Michael suggested to me that it is better to use websocket for order status updates instead of polling. So I will work on adding a coinbase user stream similar to the binance one.

Reply:
Added user stream